shell/browser/electron_browser_context.cc

```diff
@@ -431,9 +434,33 @@ void ElectronBrowserContext::SetCorsOriginAccessListForOrigin(
     std::vector<network::mojom::CorsOriginPatternPtr> allow_patterns,
     std::vector<network::mojom::CorsOriginPatternPtr> block_patterns,
     base::OnceClosure closure) {
-  // TODO(nornagon): actually set the CORS access lists. This is called from
-  // extensions/browser/renderer_startup_helper.cc.
-  base::ThreadTaskRunnerHandle::Get()->PostTask(FROM_HERE, std::move(closure));
+  auto& context_map = ElectronBrowserContext::browser_context_map();
+  auto barrier_closure =
+      BarrierClosure(1 + context_map.size(), std::move(closure));
+
+  for (auto& iter : context_map) {
+    if (iter.second) {
+      auto bc_setter = base::MakeRefCounted<content::CorsOriginPatternSetter>(
+          source_origin,
+          content::CorsOriginPatternSetter::ClonePatterns(allow_patterns),
+          content::CorsOriginPatternSetter::ClonePatterns(block_patterns),
+          barrier_closure);
+      ForEachStoragePartition(
+          std::move(iter.second.get()),
+          base::BindRepeating(&content::CorsOriginPatternSetter::SetLists,
+                              base::RetainedRef(bc_setter.get())));
+    }
+  }
+
+  // Keep the per-profile access list up to date so that we can use this to
+  // restore NetworkContext settings at anytime, e.g. on restarting the
+  // network service.
+  scoped_refptr<content::SharedCorsOriginAccessList>
+      shared_cors_origin_access_list =
+          content::SharedCorsOriginAccessList::Create();
```

Comment:
Browser context should maintain a reference to the access list and it must be published for the content layer to pick it up, three changes are needed here:

1) `shared_cors_origin_access_list` should be a member of `ElectronBrowserContext` https://github.com/electron/electron/blob/e4bcb81f47abe2651399c4d1c7763079d92f1574/shell/browser/electron_browser_context.cc#L112

2) `ElectronBrowserContext` should implement `GetSharedCorsOriginAccessList` https://github.com/electron/electron/blob/e4bcb81f47abe2651399c4d1c7763079d92f1574/shell/browser/electron_browser_context.cc#L465

3) Network context must be initialized with this list https://github.com/electron/electron/blob/e4bcb81f47abe2651399c4d1c7763079d92f1574/shell/browser/net/network_context_service.cc#L85